src/components/TodosContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { getTodos } from '../api/todos';
+import { FilterOption } from '../types/FilterOption';
+
+const initialTodos: Todo[] = [];
+
+interface TodosContextType {
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  filterOption: FilterOption,
+  setFilterOption: (filterOption: FilterOption) => void,
+  USER_ID: number,
+  errorMessage: string,
+  setErrorMessage: (errorMessage: string) => void,
+}
+
+export const TodosContext = React.createContext<TodosContextType>({
+  todos: initialTodos,
+  setTodos: () => { },
+  filterOption: FilterOption.Active,
+  setFilterOption: () => {},
+  USER_ID: 11945,
+  errorMessage: '',
+  setErrorMessage: () => { },
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [filterOption, setFilterOption] = useState<FilterOption>(
+    FilterOption.All,
+  );
+  const USER_ID = 11945;
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  const value = useMemo(() => ({
+    todos,
+    setTodos,
+    filterOption,
+    setFilterOption,
+    USER_ID,
```

Comment:
```suggestion
    userId: USER_ID,
```
generally, you can just write it out of component and export/import wherever you need
But indeed it _could_ be a dynamic variable, so you can keep it in the context, but change the object key to `camelCase`